src/h5web/vis-packs/core/heatmap/config.ts

```diff
@@ -1,62 +1,71 @@
+  setCustomDomain: (customDomain: Domain | undefined) => void;
+
   colorMap: ColorMap;
+  setColorMap: (colorMap: ColorMap) => void;
+
   scaleType: ScaleType;
+  setScaleType: (scaleType: ScaleType) => void;
+
   keepAspectRatio: boolean;
+  toggleAspectRatio: () => void;
+
   showGrid: boolean;
+  toggleGrid: () => void;
+
   autoScale: boolean;
   isAutoScaleDisabled: boolean;
+  toggleAutoScale: () => void;
+  disableAutoScale: (isAutoScaleDisabled: boolean) => void;
 }
 
-const STORAGE_CONFIG: StorageConfig = {
-  storageId: 'h5web:heatmap',
-  itemsToPersist: [
-    'colorMap',
-    'scaleType',
-    'keepAspectRatio',
-    'showGrid',
-    'autoScale',
-  ],
-};
-
-const INITIAL_STATE: HeatmapConfig = {
-  dataDomain: undefined,
-  customDomain: undefined,
-  colorMap: 'Viridis',
-  scaleType: ScaleType.Linear,
-  keepAspectRatio: true,
-  showGrid: false,
-  autoScale: false,
-  isAutoScaleDisabled: false,
-};
-
-export const useHeatmapConfig = createPersistableState(
-  STORAGE_CONFIG,
-  combine({ ...INITIAL_STATE }, (set) => ({
-    resetDomains: (dataDomain: Domain | undefined) =>
-      set({ dataDomain, customDomain: undefined }),
-
-    setCustomDomain: (customDomain: Domain | undefined) =>
-      set({ customDomain }),
-
-    setColorMap: (colorMap: ColorMap) => set({ colorMap }),
-
-    setScaleType: (scaleType: ScaleType) => set({ scaleType }),
-
-    toggleAspectRatio: () =>
-      set((state) => ({ keepAspectRatio: !state.keepAspectRatio })),
-
-    toggleGrid: () => set((state) => ({ showGrid: !state.showGrid })),
-
-    toggleAutoScale: () => set((state) => ({ autoScale: !state.autoScale })),
-
-    disableAutoScale: (isAutoScaleDisabled: boolean) =>
-      set({ isAutoScaleDisabled }),
-  }))
+export const useHeatmapConfig = create<HeatmapConfig>(
+  persist(
+    (set) => ({
+      dataDomain: undefined,
+      customDomain: undefined,
+      resetDomains: (dataDomain: Domain | undefined) =>
+        set({ dataDomain, customDomain: undefined }),
+      setCustomDomain: (customDomain: Domain | undefined) =>
+        set({ customDomain }),
+
+      colorMap: 'Viridis',
+      setColorMap: (colorMap: ColorMap) => set({ colorMap }),
+
+      scaleType: ScaleType.Linear,
+      setScaleType: (scaleType: ScaleType) => set({ scaleType }),
+
+      keepAspectRatio: true,
+      toggleAspectRatio: () =>
+        set((state) => ({ keepAspectRatio: !state.keepAspectRatio })),
+
+      showGrid: false,
+      toggleGrid: () => set((state) => ({ showGrid: !state.showGrid })),
+
+      autoScale: false,
+      isAutoScaleDisabled: false,
+      toggleAutoScale: () => set((state) => ({ autoScale: !state.autoScale })),
+      disableAutoScale: (isAutoScaleDisabled: boolean) =>
+        set({ isAutoScaleDisabled }),
+    }),
+    {
+      name: 'h5web:heatmap',
+      whitelist: [
+        'colorMap',
+        'scaleType',
+        'keepAspectRatio',
+        'showGrid',
+        'autoScale',
+      ],
+      version: 1,
```

Comment:
We just need to bump this version number the next time we modify the state, and we won't run into those nasty bugs of deprecated persisted states.